extension/docs/IntegrationGuide.md

```diff
@@ -0,0 +1,112 @@
+1. [API Extension subscription to Adyen-integration endpoints](../resources/api-extensions.json)
+1. [Custom types for payments](../resources/payment-custom-types.json)
+1. [Custom types for interface interactions](../resources/payment-interface-interaction-types.json)
+
+## Required parameters
+In order to make the extension module working, following parameters have to be provided to the Adyen-integration.
+- `CTP_PROJECT_KEY`, `CTP_CLIENT_ID`, `CTP_CLIENT_SECRET` - CTP credentials of the shop project. Go to `https://mc.commercetools.com/${your-ctp-project}/settings/developer/api-clients`. This module needs to CRUD multiple CTP resources, thus recommended scope is manage_project.
+- `API_EXTENSION_BASE_URL` - URL of the Adyen-integration. This URL will be called by CTP Extension endpoint.
+- `ADYEN_API_KEY` - Go to [Account/Users](https://ca-test.adyen.com/ca/ca/config/users.shtml) - Select a user with `Web Service` User type - Generate New API Key (notice: in case you get `403 Forbidden` error from Adyen, try to regenerate the key)   
+- `ADYEN_MERCHANT_ACCOUNT` - Go to [Account/Merchant accounts](https://ca-test.adyen.com/ca/ca/accounts/show.shtml?accountTypeCode=MerchantAccount)
+
+# Checkout steps
+1. On each checkout step [validate cart state](#validate-cart-state)
+1. Before starting payment process make sure there is no valid payments already:
+    * [Recalculate cart](#recalculate-cart)
+    * [Validate payment](#validate-payment)
+    * [Validate payment transaction](#validate-payment-transaction)
+
+If all above validations are passed then order can be created right away and order confirmation page shown.
+Otherwise shopper might continue with further payment steps.
+
+1. **Get available payment methods**  
+    1. In order to [get which payment methods are available to the current shopper](https://docs.adyen.com/developers/checkout/api-integration#step1getavailablepaymentmethods),
+    you have to create/update a CTP Payment with following properties:
+        - `Payment.paymentMethodInfo.method = empty or undefined`   
+        - `Payment.paymentMethodInfo.countryCode != null` - set the country of a shopper. Please consult with Adyen for the right format.  
+    1. Adyen-integration will make a [request](https://docs.adyen.com/developers/checkout/api-integration#step1getavailablepaymentmethods) to Adyen API.
+    1. The response will be saved in interface interaction with `type='ctp-adyen-integration-interaction'` and `fields.type='getPaymentDetails'` as stringified JSON.
+    1. Before presenting all the methods, please check Adyen-integration documentation for supported payment methods.
+1. **Continue with one of the supported payment methods:**
+    1. [Credit card payment](./CreditCardIntegration.md)  
+    1. [Paypal payment](./PaypalIntegration.md)
+
+# Error cases
+1. **Adyen returns HTTP code other than 200**  
+Request and response from Adyen are always stored in `Payment.interfaceInteractions` as strigified JSON.
```

Comment:
what is the error case in here ? can you give an example..

Reply:
Any errors from Adyen can happen. I'm not sure if it makes sense to give a random example.